test/pools/weighted/WeightedPool.test.ts

```diff
@@ -42,6 +43,188 @@ describe('WeightedPool', function () {
 
   context('for a 2 token pool (custom)', () => {
     itBehavesAsWeightedPool(2, true);
+
+    describe('oracle', () => {
+      let pool: WeightedPool, tokens: TokenList;
+
+      const weights = WEIGHTS.slice(0, 2);
+      const initialBalances = INITIAL_BALANCES.slice(0, 2);
+
+      type PoolHook = (lastChangeBlock: number) => Promise<unknown>;
+
+      sharedBeforeEach('deploy pool', async () => {
+        tokens = allTokens.subset(2);
+        const params = { twoTokens: true, tokens, weights, swapFeePercentage: POOL_SWAP_FEE_PERCENTAGE };
+        pool = await WeightedPool.create(params);
+      });
+
+      const initializePool = () => {
+        sharedBeforeEach('initialize pool', async () => {
+          await pool.init({ initialBalances, recipient: lp });
+        });
+      };
+
+      const calcLastChangeBlock = async (offset: number): Promise<number> => {
+        const nextBlockNumber = (await lastBlockNumber()) + 1;
+        return nextBlockNumber - offset;
+      };
+
+      const itUpdatesTheOracleData = (action: PoolHook, lastChangeBlockOffset = 0) => {
+        it('updates the oracle data', async () => {
+          const previousData = await pool.instance.miscData();
+
+          await advanceTime(MINUTE * 10); // force index update
+          await action(await calcLastChangeBlock(lastChangeBlockOffset));
+
+          const currentMiscData = await pool.instance.miscData();
+          expect(currentMiscData.oracleIndex).to.equal(previousData.oracleIndex.add(1));
+          expect(currentMiscData.oracleSampleInitialTimestamp).to.equal(await currentTimestamp());
+        });
```

Comment:
Hm, I was thinking of having another `it` here that doesn't advance time and checks the update, but that'd require these tests knowing too much about the update procedure.